test/route_end_to_end_connectivity_test.cc

```diff
@@ -0,0 +1,291 @@
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <gtest/gtest.h>
+#include <maliput/api/compare.h>
+#include <maliput/api/lane.h>
+#include <maliput/api/lane_data.h>
+#include <maliput/api/regions.h>
+#include <maliput/api/road_geometry.h>
+#include <maliput/api/road_network.h>
+#include <maliput/common/filesystem.h>
+#include <maliput/common/maliput_throw.h>
+#include <maliput/routing/phase.h>
+#include <maliput/routing/route.h>
+#include <maliput_malidrive/builder/params.h>
+#include <maliput_malidrive/builder/road_network_builder.h>
+#include <maliput_malidrive/loader/loader.h>
+
+#include "assert_compare.h"
+
+// @file This file evaluates the end to end connectivity constraints of a maliput::routing::Route.
+// @details This is not done in maliput because of the need of a concrete maliput::api::RoadGeometry
+// to evaluate all the topological and geometrical constraints. Mocking that for the purposes of
+// this test would require implementing a concrete backend. In this test, an ad-hoc xodr map is used.
+// We don't need any fancy geometry or network topology, just two roads with two lanes each are enough
+// to validate the end to end connectivity is correct.
+
+namespace maliput {
+namespace test {
+namespace {
+
+// Converts @p errors into a single string by concatenating each item with a " | " as separation.
+// @param errors A vector of strings.
+// @return A string with all items in @p error concatenated with " | " as separation.
+std::string SerializeErrors(const std::vector<std::string>& errors) {
+  std::string result;
+  for (const std::string& error : errors) {
+    result += error + " | ";
+  }
+  return result;
+}
+
+class MultipleStraightSegmentsTest : public ::testing::Test {
+ public:
```

Comment:
To be consistent with other tests in this directory, mark this `protected`.

Reply:
[5415ba7](https://github.com/maliput/maliput_integration_tests/pull/81/commits/5415ba7b4a80d037e1a4fe16999f64b6bfd528aa)